pkg/cable/libreswan/libreswan.go

```diff
@@ -584,3 +610,51 @@ func (i *libreswan) runPluto() error {
 
 	return nil
 }
+
+func (i *libreswan) addClampMssToPathMTU(subnets string) {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+	ruleSpecDest := []string{"-d", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+
+	if err := i.updateMangleTableEntryWithRetry(i.ipt.AppendUnique, ruleSpecSource...); err != nil {
+		klog.Errorf("error appending iptables rule \"%s\": %v\n", strings.Join(ruleSpecSource, " "), err)
+	}
+
+	if err := i.updateMangleTableEntryWithRetry(i.ipt.AppendUnique, ruleSpecDest...); err != nil {
+		klog.Errorf("error appending iptables rule \"%s\": %v\n", strings.Join(ruleSpecDest, " "), err)
+	}
+}
+
+func (i *libreswan) removeClampMssToPathMTU(subnets string) {
+	ruleSpecSource := []string{"-s", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+	ruleSpecDest := []string{"-d", subnets, "-p", "tcp", "-m", "tcp", "--tcp-flags", "SYN,RST", "SYN", "-j",
+		"TCPMSS", "--clamp-mss-to-pmtu"}
+
+	if err := i.updateMangleTableEntryWithRetry(i.ipt.Delete, ruleSpecSource...); err != nil {
+		klog.Errorf("error deleting iptables rule \"%s\": %v\n", strings.Join(ruleSpecSource, " "), err)
+	}
+
+	if err := i.updateMangleTableEntryWithRetry(i.ipt.Delete, ruleSpecDest...); err != nil {
+		klog.Errorf("error deleting iptables rule \"%s\": %v\n", strings.Join(ruleSpecDest, " "), err)
+	}
+}
+
+func (i *libreswan) updateMangleTableEntryWithRetry(updater func(table, chain string, rulespec ...string) error, rulespec ...string) error {
+	retryConfig := wait.Backoff{
+		Cap:      2 * time.Minute,
+		Duration: 1 * time.Second,
+		Factor:   2,
+		Steps:    4,
+	}
+
+	err := retry.OnError(retryConfig, func(err error) bool {
+		return true
+	}, func() error {
+		err := updater(mangleTable, subMarinerPostRoutingChain, rulespec...)
+		return err
+	})
```

Comment:
As indicated earlier, I think it would be better to propagate the error and let the cable engine handle retry async so we don't block it. In that case, we don't need really this function.

Reply:
Done.